### ISSUE TITLE
Regarding number of train test samples

### ISSUE THREAD

**Issue by Sreelakshmi-k** [ThilinaRajapakse/pytorch-transformers-classification#33]
- **My train file has 8000 sentences but when i implemented this code it shows number of samples =817**

INFO:__main__:Creating features from dataset file at data/
8000
817
100%|██████████| 817/817 [00:01<00:00, 537.09it/s]
INFO:__main__:Saving features into cached file data/cached_train_bert-base-multilingual-cased_128_binary
INFO:__main__:***** Running training *****
INFO:__main__:  Num examples = 817
INFO:__main__:  Num Epochs = 35
INFO:__main__:  Total train batch size  = 8
INFO:__main__:  Gradient Accumulation steps = 1
INFO:__main__:  Total optimization steps = 3605

- **Similarly my test file has 2000 sentences but when evaluation code was executed it showed num examples =18**

INFO:__main__:Evaluate the following checkpoints: ['outputs/checkpoint-2000', 'outputs']
INFO:__main__:Creating features from dataset file at data/
2000
18
100%|██████████| 18/18 [00:00<00:00, 148.27it/s]
INFO:__main__:Saving features into cached file data/cached_dev_bert-base-multilingual-cased_128_binary
INFO:__main__:***** Running evaluation 2000 *****
INFO:__main__:  Num examples = 18
INFO:__main__:  Batch size = 8
Evaluating
100% 3/3 [00:00<00:00, 7.02it/s]
INFO:__main__:***** Eval results 2000 *****
INFO:__main__:  fn = 4
INFO:__main__:  fp = 3
INFO:__main__:  mcc = 0.20385887657505022
INFO:__main__:  tn = 7
INFO:__main__:  tp = 4

INFO:__main__:Loading features from cached file data/cached_dev_bert-base-multilingual-cased_128_binary
INFO:__main__:***** Running evaluation outputs *****
INFO:__main__:  Num examples = 18
INFO:__main__:  Batch size = 8
Evaluating
100% 3/3 [00:00<00:00, 7.51it/s]

INFO:__main__:***** Eval results outputs *****
INFO:__main__:  fn = 4
INFO:__main__:  fp = 2
INFO:__main__:  mcc = 0.31622776601683794
INFO:__main__:  tn = 8
INFO:__main__:  tp = 4

- **also the final output obtained is**

 {'fn_2000': 4,
 'fn_outputs': 4,
 'fp_2000': 3,
 'fp_outputs': 2,
 'mcc_2000': 0.20385887657505022,
 'mcc_outputs': 0.31622776601683794,
 'tn_2000': 7,
 'tn_outputs': 8,
 'tp_2000': 4,
 'tp_outputs': 4}

**

- **But in total my test_df has 2000 sentences , but wen i add tp+tn+fp+fn i only get 18. Could you please explain this.**

**

**Comment by ThilinaRajapakse** [comment]
The data might be being loaded from the cache dir. Try deleting any cached files.

Do you have the same issue when using the yelp data?

**Comment by Sreelakshmi-k** [comment]
I haven't tried the any other data

**Comment by Sreelakshmi-k** [comment]
i have provided the right location from where my data has to be take. I
have kept both test and train files in a folder called data which is in my
drive. Could u please tell  me what i should do to train the whole 8000
data and test whole 2000
Regards,
Sreelakshmi

**Comment by ThilinaRajapakse** [comment]
Try using the Yelp dataset as given in the guide. It's impossible to say what the issue is without seeing your data.

Or, consider using [Simple Transformers](https://github.com/ThilinaRajapakse/simpletransformers) as it is up to date and much easier to use.

**Comment by Sreelakshmi-k** [comment]
I saw that you yelp data had a column of texts and a column of labels. My
data is also the same. I will try simple transformer
